rpcd/playbooks/roles/elasticsearch/tasks/upgrade_pre_flight.yml

```diff
@@ -0,0 +1,48 @@
+---
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+- name: Install Pre Flight Checks
+  template:
+    src: upgrade-pre-flight-checks.py.j2
+    dest: /opt/upgrade-pre-flight-checks.py
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - es-pre-flight
+
+- name: Run Pre Flight Checks
+  shell: /opt/upgrade-pre-flight-checks.py
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  register: pre_flight_status
+  ignore_errors: true
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - es-pre-flight
+
+- name: Check for latest version
+  set_fact:
+    logging_upgrade: False
+  when:
+    - pre_flight_status.rc == '1'
+
+- name: Check for reindexing in progress
+  command: /bin/false
```

Comment:
the return codes should be ints but ansible is really funky with type checking.  I just got rid of that as we can't not run the upgrade plays if the version is already at the latest, there isn't a way that I can see to get that fact back up into the main.yml playbook.